packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -511,15 +518,18 @@ export class GraphqlApi extends GraphqlApiBase {
 
   private setupLogConfig(config?: LogConfig) {
     if (!config) return undefined;
-    const role = new Role(this, 'ApiLogsRole', {
-      assumedBy: new ServicePrincipal('appsync.amazonaws.com'),
-      managedPolicies: [
-        ManagedPolicy.fromAwsManagedPolicyName(
-          'service-role/AWSAppSyncPushToCloudWatchLogs'),
-      ],
-    });
+    let apiLogsRoleArn = config.roleArn;
+    if (!apiLogsRoleArn) {
+      const role = new Role(this, 'ApiLogsRole', {
+        assumedBy: new ServicePrincipal('appsync.amazonaws.com'),
+        managedPolicies: [
+          ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSAppSyncPushToCloudWatchLogs'),
+        ],
+      });
+      apiLogsRoleArn = role.roleArn;
+    }
```

Comment:
```suggestion
let logsRoleArn = config.roleArn ?? new Role(this, 'ApiLogsRole', {
  assumedBy: new ServicePrincipal('appsync.amazonaws.com'),
  managedPolicies: [
    ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSAppSyncPushToCloudWatchLogs'),
  ],
}).roleArn;
```

Reply:
Sure. Wanted to ensure readability.

Think that would allow us to use `const logsRoleArn` as well.